docs/DeveloperGuide.md

```diff
@@ -303,25 +363,57 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
   Use case ends.
 
 * 3a. The given index is invalid.
+    * 3a1. Duke The Market shows an error message.
+
+      Use case resumes at step 3.
+
+
+**UC04: Export email addresses of customers to a file**
+
+**MSS**
+1.  User requests to export to a specified file the email address of customers who attended a specific event.
+2.  Duke The Market exports the customers’ email addresses to the specified file
+
+    Use case ends.
+
+
+**UC05: Retrieve customers who may be interested in an upcoming event**
+
+**MSS**
+1. User requests to list events
+2. Duke The Market shows a list of events
+3. User requests to show customers who may be interested in an upcoming event in the list
+4. Duke The Market shows the customers who may be interested in that upcoming event
+5. User requests to export the customers' email addresses to a specified file (UC04)
+
+   Use case ends.
+
 
-    * 3a1. AddressBook shows an error message.
 
-      Use case resumes at step 2.
 
 *{More to be added}*
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+2. App should not require installation to work, i.e. it is portable
+3. App should work well for standard screen resolutions 1920x1080 and higher, as well as for screen scales 100% and 125%.
+4. App should be usable (i.e., all functions can be used even if the user experience is not optimal) for resolutions 1280x720 and higher, as well as for screen scales 150%.
+5. Should be able to hold up to 1000 persons and 200 events without a noticeable sluggishness in performance for typical usage.
+6. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+7. The system should respond within 5 seconds.
+8. Data should be stored locally in a human editable text file.
+9. Data should not be stored using an external database management system (DBMS)
+10. File size of the data should not exceed 1GB (should be stored in a space saving manner)
+11. App should be an executable (Double clicked or can be run using the “java -jar” command).
```

Comment:
Perhaps format `java -jar` as code since it's a command?

```suggestion
11. App should be an executable (Double clicked or can be run using the `java -jar` command).
```